pkg/monitoring/InfluxDbStorage.php

```diff
@@ -158,21 +183,44 @@ public function pushSentMessageStats(SentMessageStats $stats): void
             new Point($this->config['measurementSentMessages'], 1, $tags, [], $stats->getTimestampMs()),
         ];
 
-        $this->getDb()->writePoints($points, Database::PRECISION_MILLISECONDS);
+        $this->doWrite($points);
     }
 
-    private function getDb(): Database
+    private function doWrite(array $points): void
     {
-        if (null === $this->database) {
-            if (null === $this->client) {
-                $this->client = new Client(
-                    $this->config['host'],
-                    $this->config['port'],
-                    $this->config['user'],
-                    $this->config['password']
-                );
+        if (!$this->client || $this->client->getDriver() instanceof QueryDriverInterface) {
+            $this->getDb()->writePoints($points, Database::PRECISION_MILLISECONDS, $this->config['retentionPolicy']);
+        } else {
+            // Code below mirrors what `writePoints` method of Database does.
```

Comment:
I was wondering if maybe we should just write to `Client` always instead. 

I'll probably propose to move `writePayload` method into `Client` instance in `influxdb-php`, since `Client` is called anyway and overall `Client` should know how to handle it's parameters when writing.

Reply:
Should we wait with this PR it a PR to influxdb-php is merged?

Reply:
Honestly influxdb-php maintainers respond kinda slow and it will probably take some time before any such change is merged into it :)
My original PR for it (with TCP driver) does not change the API in any way, and the one I've suggested above will change their API (adding a new method to `Client`). So I'd say don't wait for it?
If such a change is introduced to influxdb-php it will mean that php-enqueue will have either maintain this fallback for a while or only allow influxdb-php versions that contain the fix.

Reply:
I am fine to get it merged, one question: do we really need to deprecate passing a client as an argument. As for me, it still a valid option.

Reply:
I'd suggest deprecating it, because it basically prevents setting any other options, like database name, `measurementSentMessages`, `measurementConsumedMessages` and `measurementConsumers` (others being basically configuration options for the client itself).

In my opinion it would be best to be as strict as possible when creating function definitions, in this case a constructor. If there is a secondary constructor needed, it can always take the form of a static function, like so:
```php
public static function createFromClient(Client $client, array $config): self
{
    $self = new static($config);
    $self->client = $client;
    return $self;
}
```
or similar. This would allow us to skip most of type checks, as PHP will do a function argument check on it's own and function declaration would explicitly say what value types are acceptable.

I'll drop the deprecation for now anyway.

Reply:
okay, let's do it.

Reply:
Should I revert the last commit then? :)

Reply:
I'll make a part of another PR maybe, so this could be merged and anything else regarding constructor parameters and so on can be left for discussion.

Reply:
You can add a static create method as part of this PR and let's deprecate client arg. No need for extra PR.

Reply:
Commited. 

I've taken the liberty to convert `parseDsn` method into a static function, since it did not make use of `$this`, which allowed me to use it inside `createWithClient` static function.

I've noticed that there is a possibility of array configuration values to become overwritten, if `dsn` array element is passed, but I'm leaving it as is. From what I understand, it's possible that Symfony bundle might be passing `$config` with `dsn` field which would break if removed (?).

Otherwise I think it's good to go, unless you want some other changes as well?